ktx/src/main/java/permissions/dispatcher/ktx/PermissionsRequesterImpl.kt

```diff
@@ -1,39 +1,81 @@
+import androidx.fragment.app.FragmentManager
+import androidx.lifecycle.LifecycleOwner
 import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.ViewModelStoreOwner
 import permissions.dispatcher.PermissionUtils
 
 internal class PermissionsRequesterImpl(
     private val permissions: Array<out String>,
-    private val activity: FragmentActivity,
+    private val view: ViewType,
     private val onShowRationale: ShowRationaleFun?,
     private val onPermissionDenied: Fun?,
     private val requiresPermission: Fun,
     private val onNeverAskAgain: Fun?,
     private val permissionRequestType: PermissionRequestType
 ) : PermissionsRequester {
     override fun launch() {
-        if (permissionRequestType.checkPermissions(activity, permissions)) {
+        if (permissionRequestType.checkPermissions(view.context, permissions)) {
             requiresPermission()
         } else {
-            ViewModelProvider(activity).get(PermissionRequestViewModel::class.java).observe(
-                activity,
-                requiresPermission,
-                onPermissionDenied,
-                onNeverAskAgain
-            )
+            ViewModelProvider(view.viewModelStoreOwner).get(PermissionRequestViewModel::class.java)
+                .observe(
+                    view.lifeCycleOwner,
+                    requiresPermission,
+                    onPermissionDenied,
+                    onNeverAskAgain
+                )
             val requestFun = {
-                activity.supportFragmentManager
+                view.fragmentManager
                     .beginTransaction()
                     .replace(android.R.id.content, permissionRequestType.fragment(permissions))
                     .commitNowAllowingStateLoss()
             }
-            if (PermissionUtils.shouldShowRequestPermissionRationale(activity, *permissions)) {
+            if (shouldShowRequestPermissionRationale(view, permissions)) {
                 onShowRationale?.invoke(KtxPermissionRequest.create(onPermissionDenied, requestFun))
             } else {
                 requestFun.invoke()
             }
         }
     }
+
+    private fun shouldShowRequestPermissionRationale(
+        view: ViewType,
+        permissions: Array<out String>
+    ): Boolean = when (view) {
+        is ActivityType -> PermissionUtils.shouldShowRequestPermissionRationale(
+            view.left,
+            *permissions
+        )
+        is FragmentType -> PermissionUtils.shouldShowRequestPermissionRationale(
+            view.right,
+            *permissions
+        )
+    }
 }
+
+private val ViewType.context: Context
+    get() = when (this) {
+        is ActivityType -> left.applicationContext
+        is FragmentType -> right.requireContext()
+    }
+
+private val ViewType.viewModelStoreOwner: ViewModelStoreOwner
+    get() = when (this) {
+        is ActivityType -> left
+        is FragmentType -> right
+    }
+
+private val ViewType.lifeCycleOwner: LifecycleOwner
```

Comment:
```suggestion
private val ViewType.lifecycleOwner: LifecycleOwner
```

For consistency with the AndroidX property ☀️

Reply:
thx bro! I'll think something over entirely and polish PR with description😄